library/network/src/modules/NetworkService.rb

```diff
@@ -93,18 +94,15 @@ def main
     # Helper to run systemctl actions
     # @return exit code
     def RunSystemCtl(service, action)
-      cmd = Builtins.sformat("%1 %2 %3.service", SYSTEMCTL, action, service)
-      ret = Convert.convert(
-        SCR.Execute(path(".target.bash_output"), cmd, "TERM" => "raw"),
-        from: "any",
-        to:   "map <string, any>"
-      )
+      cmd = Builtins.sformat("%1 %2 %3.service", SYSTEMCTL.shellescape,
```

Comment:
Having a constant SYSTEMCTL in that class and then needing to shell-escape that here is IMHO very much counterproductive and makes the code less readable and maintainable, as we can see here: It does not need to be shell-escaped because it's a constant, and it is used for a grand total of **one** (1) time in the complete code.

Please change this to simply use

```Ruby
cmd = "/usr/bin/systemctl #{action.shellescape} #{service.shellescape}"

and get rid of the constant in the class header.
```